pkg/controller/replicaset/replica_set.go

```diff
@@ -671,9 +694,8 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	// Always updates status as pods come up or die.
 	if err := updateReplicaCount(rsc.kubeClient.Extensions().ReplicaSets(rs.Namespace), rs, len(filteredPods), fullyLabeledReplicasCount, readyReplicasCount); err != nil {
 		// Multiple things could lead to this update failing. Requeuing the replica set ensures
-		// we retry with some fairness.
-		glog.V(2).Infof("Failed to update replica count for controller %v/%v; requeuing; error: %v", rs.Namespace, rs.Name, err)
-		rsc.enqueueReplicaSet(&rs)
+		// Returning an error causes a requeue without forcing a hotloop
+		errlist = append(errlist, err)
 	}
-	return nil
+	return utilerrors.NewAggregate(errlist)
```

Comment:
Actually there is no difference on a second look I had. Still I would prefer to follow the same pattern as the RC controller and not aggregate here.